src/store/actions/commentsActions.js

```diff
@@ -0,0 +1,82 @@
+import { toast } from 'react-toastify';
+import axiosConfig from '../../axiosConfig';
+import {
+  FETCH_AN_ARTICLE,
+  FETCH_ARTICLE_COMMENTS,
+  FETCH_AN_ARTICLE_404,
+  CREATE_COMMENT_SUCCESS,
+  CREATE_REPLY_SUCCESS,
+} from './actionTypes';
+
+const getItem = (actionType, data) => ({
+  type: actionType,
+  payload: data,
+});
+
+export const fetchAnArticle = slug => (dispatch) => {
+  return axiosConfig
+    .request({
+      method: 'get',
+      url: `articles/${slug}/`,
+    })
+    .then((response) => {
+      dispatch({
+        type: FETCH_AN_ARTICLE,
+        payload: response.data.Article,
+      });
+    })
+    .catch((error) => {
+      (getItem(FETCH_AN_ARTICLE_404, error));
+    });
+};
+
+export const fetchArticleComments = slug => (dispatch) => {
+  return axiosConfig
+    .request({
+      method: 'get',
+      url: `articles/${slug}/comments/`,
+    })
+    .then((response) => {
+      dispatch(getItem(FETCH_ARTICLE_COMMENTS, response.data.Comments));
+    })
+    .catch((error) => {
+      toast.error(error, {
+        position: toast.POSITION.BOTTOM_LEFT,
+      });
+    });
+};
+
+export const createAComment = (slug, data) => (dispatch) => {
+  return axiosConfig
+    .request({
+      method: 'post',
+      url: `articles/${slug}/comments/`,
+      data: { body: data },
+    })
+    .then((response) => {
+      console.log(response.data);
+      dispatch(getItem(CREATE_COMMENT_SUCCESS, response.data));
+    })
+    .catch((error) => {
+      toast.error(error, {
+        position: toast.POSITION.BOTTOM_LEFT,
+      });
+    });
+};
+
+export const createAReply = (slug, pk, data) => (dispatch) => {
+  return axiosConfig
+    .request({
+      method: 'post',
+      url: `articles/${slug}/comments/${pk}/`,
+      data: { body: data },
+    })
+    .then(() => {
+      dispatch(getItem({ type: CREATE_REPLY_SUCCESS }));
+    })
+    .catch((error) => {
+      toast.error(error, {
```

Comment:
Please dispatch a failure action to the store here. Something similar to `CREATE_REPLY_FAILURE`